server/script/core/code-action.lua

```diff
@@ -137,7 +142,222 @@ local function changeVersion(uri, version, results)
+    end
+
+    return table.concat(builder)
+end
+
+local function getAbsoluteLuaPath(path)
+    local builder = { path[1].name }
+
+    for index, node in ipairs(path) do
+        if index ~= 1 then
+            if not canIndexObject(node) then
+                return
+            end
+
+            table.insert(builder, getSafeIndexer(node.name))
+        end
+    end
+
+    return table.concat(builder)
+end
+
+local function solveUndefinedGlobalImport(uri, diag, results)
+    local ast    = files.getAst(uri)
+    local offset = files.offsetOfWord(uri, diag.range.start)
+    local name = guide.eachSourceContain(ast.ast, offset, function (source)
+        if source.type ~= 'getglobal' then
+            return
+        end
+
+        return guide.getKeyName(source)
+    end)
+
+    local rawMatches = findMatchingScripts(name)
+    if rawMatches[1] == nil then
+        return
+    end
+
+    local sourcePath = findPath(uri)
+    
+    local matches = {}
+    for _, match in ipairs(rawMatches) do
+        -- Never do `require(script)` or equivalent.
+        if match.object.uri ~= uri then
+            local targetPath = match.path
+            match.relativeLuaPath = sourcePath and findRelativeLuaPath(sourcePath, targetPath)
+            match.absoluteLuaPath = getAbsoluteLuaPath(targetPath)
+            -- If the path tries to index into a script and that's disallowed,
+            -- it won't have any paths available
+            if match.relativeLuaPath or match.absoluteLuaPath then
+                table.insert(matches, match)
+            end
+        end
+    end
+    
+    -- Relative-available matches first, sorted by smallest lua path string,
+    -- followed by absolute paths, sorted by smallest lua path string
+    table.sort(matches, function(a, b)
+        if a.relativeLuaPath and b.relativeLuaPath then
+            return a.relativeLuaPath < b.relativeLuaPath
+        elseif a.relativeLuaPath or b.relativeLuaPath then
+            return a.relativeLuaPath ~= nil
+        else
+            return a.absoluteLuaPath < b.absoluteLuaPath
+        end
+    end)
+
+    for index, match in ipairs(matches) do
+        -- Don't display too many results if we found many matches
+        if index > 10 then
+            break
+        end
+
+        local luaPath = match.relativeLuaPath or match.absoluteLuaPath
+
+        results[#results+1] = {
+            title = lang.script('ACTION_IMPORT_SUGGESTED', luaPath),
+            kind = 'quickfix',
+            edit = {
+                changes = {
+                    [uri] = {
+                        {
+                            start   = 1,
+                            finish  = 0,
+                            newText = ('local %s = require(%s)\n'):format(name, luaPath),
+                        },
+                    }
+                }
+            }
+        }
+    end
+end
+
 local function solveUndefinedGlobal(uri, diag, results)
+    solveUndefinedGlobalImport(uri, diag, results)
```

Comment:
Move this above [`markGlobal(uri, name, results)`](https://github.com/Corecii/RobloxLsp/blob/e16862e97b09ce933cf5aed3673eff47138c746c/server/script/core/code-action.lua#L368) and pass `name` as an argument.

Reply:
I've made this its own diagnostic now, so this suggestion is no longer relevant 👍